pages/integrations/index.tsx

```diff
@@ -2,29 +2,82 @@ import Head from 'next/head';
 import Link from 'next/link';
 import prisma from '../../lib/prisma';
 import Shell from '../../components/Shell';
-import {useState} from 'react';
+import {useEffect, useState} from 'react';
 import {getSession, useSession} from 'next-auth/client';
-import {CheckCircleIcon, ChevronRightIcon, PlusIcon, XCircleIcon} from '@heroicons/react/solid';
+import {CalendarIcon, CheckCircleIcon, ChevronRightIcon, PlusIcon, XCircleIcon} from '@heroicons/react/solid';
 import {InformationCircleIcon} from '@heroicons/react/outline';
+import { Switch } from '@headlessui/react'
 
 export default function Home({ integrations }) {
     const [session, loading] = useSession();
     const [showAddModal, setShowAddModal] = useState(false);
-
-    if (loading) {
-        return <p className="text-gray-400">Loading...</p>;
-    }
+    const [showSelectCalendarModal, setShowSelectCalendarModal] = useState(false);
+    const [selectableCalendars, setSelectableCalendars] = useState([]);
 
     function toggleAddModal() {
         setShowAddModal(!showAddModal);
     }
 
+    function toggleShowCalendarModal() {
+        setShowSelectCalendarModal(!showSelectCalendarModal);
+    }
+
+    function loadCalendars() {
+        fetch('api/availability/calendar')
+          .then((response) => response.json())
+          .then(data => {
+              setSelectableCalendars(data)
+          });
```

Comment:
We should consider introducing a global error handler to catch rejected promises and display them to the user / catch and ignore them if that's fine. Or at least use `.catch((err) => do whatever you want with err);` somewhere in your Promise chain after `fetch(...)`. For now (not a stable version yet), I'd say it's fine as you did it, but I think for the future that would be neat. Or what do you think @baileypumfleet @emrysal @PeerRich?